app/Http/Controllers/DepartmentController.php

```diff
@@ -51,28 +96,86 @@ public function show(string $id)
     /**
      * Show the form for editing the specified resource.
      */
-    public function edit(string $id)
+    public function edit(Request $request, string $id)
     {
-        $department = Department::findOrFail($id);
-        return view('departments.edit', [
-            'department' => $department
-        ]);
+        if($request->user()->isAdmin())
+        {
+            $department = Department::findOrFail($id);
+            $sectors = Sector::all();
+            return view('departments.edit', [
+                'department' => $department,
+                'sectors' => $sectors
+            ]);
+        }
+        else
+        {
+            abort(401);
+        }
     }
 
     /**
      * Update the specified resource in storage.
      */
-    public function update(Request $request, string $id)
+    public function update(Request $request, string $id) : RedirectResponse
+    {
+        // Validate the incoming request data
+        $validated = $request->validate([
+            'name' => ['required','string','max:255'], // required string, max len 255
+            'description' => ['nullable','string','max:255'],  // optional string, max len 255
+            'sector_id' => ['required','integer','exists:sectors,id']     // Ensure 'sector' is a valid integer and exists in the sectors table
+        ]);
+
+        // Find the department by ID
+        $department = Department::findOrFail($id);
+
+        // Update the department profile with the validated data
+        $department->update($validated);
+
+        // Optionally, flash a success message to the session
+        return redirect()->route('departments.index')
+            ->with('success', [
+                'message' => "Department <span class=\"text-brand-green\">{$department->name}</span> updated successfully",
+                'action_text' => 'View Department',
+                'action_url' => route('departments.show', $department->id),
+            ]);
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     */
+    public function delete(Request $request, string $id): View
     {
-        //
+        if($request->user()->isAdmin())
```

Comment:
Refactor to use `Auth::user` facade.

Reply:
resolved in 59ff09f7743b739de37cce6a45b1e3a7053d17fb